### DISCUSSION TITLE
Allow functions like `hide` to be called directly in math-mode

### DISCUSSION THREAD

**Discussion by NicoWeio** [typst/typst#2272] [Ideas]
Currently, one has to write `phi^#hide[$*$]`, which is significantly more cumbersome to write than `phi^hide(x)` (if one uses it a lot) and made me doubt that the layout still works as expected (but it seems to). (Note: Currently, the latter expression would give `unknown variable: hide`.)

I assume this applies to a few more functions, but `hide` is the one I just thought of.

Is there any disadvantage in making such functions “math-native”?

**Comment by Dherse** [comment]
I would advise opening of an issue on this (if there is not one already) since feature requests should ideally go there ;) However, I do agree with you and I think it makes sense, mind you there may be some reason why it's not the case I am simply unaware of.

**Comment by Jollywatt** [comment]
Why this works I don’t quite grok, but you can do this with:

```
#let hide = it => hide(it)
$f(x, hide(y)) = z$
```
